pkg/api/extensions/accessor_test.go

```diff
@@ -112,6 +112,14 @@ var _ = Describe("Accessor", func() {
 					Expect(acc.GetProviderConfig()).To(BeNil())
 				})
 			})
+
+			Describe("#GetExtensionPurpose", func() {
+				It("should return nil", func() {
+					acc := mkUnstructuredAccessorWithSpec(extensionsv1alpha1.DefaultSpec{})
+
+					Expect(acc.GetExtensionPurpose()).To(BeNil())
+				})
```

Comment:
What about the test for non-nil case?

Reply:
Can't be implemented, test is based on `DefaultSpec` which doesn't have field `purpose`.

Reply:
How about
```
				It("should return purpose", func() {
					purpose := extensionsv1alpha1.Normal
					acc := mkUnstructuredAccessor(&extensionsv1alpha1.ControlPlane{Spec: extensionsv1alpha1.ControlPlaneSpec{Purpose: &purpose}})

					Expect(acc.GetExtensionSpec().GetExtensionPurpose()).To(PointTo(Equal(string(purpose))))
				})
```
?